core/src/main/java/org/elasticsearch/index/indexing/ShardIndexingService.java

```diff
@@ -124,19 +128,31 @@ public void postCreate(Engine.Create create) {
             try {
                 listener.postCreate(create);
             } catch (Exception e) {
-                logger.warn("post listener [{}] failed", e, listener);
+                logger.warn("postCreate listener [{}] failed", e, listener);
             }
         }
     }
 
     public void postCreate(Engine.Create create, Throwable ex) {
+        for (IndexingOperationListener listener : listeners) {
+            try {
+                listener.postCreate(create, ex);
+            } catch (Exception e) {
+                logger.warn("postCreate listener [{}] failed", e, listener);
+            }
+        }
     }
 
     public Engine.Index preIndex(Engine.Index index) {
         totalStats.indexCurrent.inc();
         typeStats(index.type()).indexCurrent.inc();
         for (IndexingOperationListener listener : listeners) {
-            index = listener.preIndex(index);
+            try {
+                listener.preIndex(index);
+            } catch (Exception e) {
```

Comment:
@s1monw hmm Index changes already commited after Index action? If we fail in listener in post, document will be commited by lucene, but ES will think it is not indexed. yes? If yes we should implement try catch in post actions...